Sources/SwiftSyntaxBuilder/Trivia+Extension.swift

```diff
@@ -0,0 +1,37 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2022 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import SwiftSyntax
+
+extension Trivia {
+  /// Creates a new `Trivia` and adds spacing if needed.
+  ///
+  /// If there is 3 or more, then we see if the first two is a newline and a space
+  /// If so, we need to replace add the correct amount of space after each newline
+  /// comming after to have the right indentation of the code.
+  func addSpacingAfterNewlinesIfNeeded() -> Trivia {
+    var updatedTriviaPieces = self.pieces
+    if updatedTriviaPieces.count > 2,
+       case .newlines = updatedTriviaPieces[0],
+       case .spaces(let spaces) = updatedTriviaPieces[1] {
```

Comment:
What about indentation with tabs? 😬

Reply:
Ohhh, you are a dare devil 😂

https://www.youtube.com/watch?v=SsoOG6ZeyUI